plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/session/CleaningJobRegistry.java

```diff
@@ -0,0 +1,65 @@
+package org.obeonetwork.m2doc.sirius.session;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.util.List;
+import java.util.Map;
+
+import org.obeonetwork.m2doc.genconf.Generation;
+
+/**
+ * Register of cleaning jobs.
```

Comment:
Couldn't you use the plug-in instance to do the same instead of introducing a new class?